lib/http/form_data/multipart.rb

```diff
@@ -16,10 +16,8 @@ class Multipart
 
       # @param [#to_h, Hash] data form data key-value Hash
       def initialize(data, boundary: self.class.generate_boundary)
-        parts = Param.coerce FormData.ensure_hash data
-
         @boundary = boundary.to_s.freeze
-        @io = CompositeIO.new [*parts.flat_map { |part| [glue, part] }, tail]
+        @io = CompositeIO.new [*parts(data).flat_map { |part| [glue, part] }, tail]
```

Comment:
```suggestion
        @io = CompositeIO.new(parts(data).flat_map { |part| [glue, part] } << tail)
```

Reply:
Also for performance reasons, I propose to get rid of `Param.coerce` completely :D

Reply:
It shall be done.